TriangleNet_Engine/Compute/DelaunayTriangulation.cs

```diff
@@ -136,6 +139,93 @@ public static List<Polyline> DelaunayTriangulation(this Polyline outerCurve, Lis
 
             return meshPolylines;
         }
+
+        /***************************************************/
+
+        [Description("Creates a list of closed Delaunay polyline cells from a list of coplanar points. Each cell will correspond to a face in a Delaunay mesh.")]
+        [Input("points", "The points to connect to Delanuay cells. All points need to be coplanar ")]
+        [Input("plane", "Optional plane of the points. Needs to be coplanar with all of the points. If no plane is provided, a best fit plane through the points will be used.")]
+        [Input("tolerance", "Numeric tolerance to be used by the method.", typeof(Length))]
+        [Output("curves", "List of closed polyline representing faces in a Delaunay.")]
+        public static List<Polyline> DelaunayTriangulation(List<Point> points, Plane plane = null, double tolerance = Tolerance.Distance)
+        {
+            //Preform check all inputs that triangulation can be done
+            if (points == null || points.Count < 3)
+            {
+                Reflection.Compute.RecordError("Insuffient points for triangulation. Please provide at least 3 Points!");
+                return new List<Polyline>();
+            }
+
+            if (points.IsCollinear(tolerance))
+            {
+                Reflection.Compute.RecordError("Points are colinear and can not be triangulated.");
+                return new List<Polyline>();
+            }
+
+            //Basic edge case, simply create a triangle
+            if (points.Count == 3)
+            {
+                Polyline pLine = new Polyline { ControlPoints = points };
+                pLine.ControlPoints.Add(points[0]);
+                return new List<Polyline> { pLine };
+            }
+
+            //Try fit plane if no is provided
+            if (plane == null)
+            {
+                plane = points.FitPlane(tolerance);
+            }
+
+            if (plane == null)
+            {
+                Engine.Reflection.Compute.RecordError("Could not fit a plane through the Points.");
```

Comment:
Worth expanding this error message to include `and no plane was provided` so the user knows that providing their own plane is an option?